app/middleware/postcodeLookup.js

```diff
@@ -8,48 +7,41 @@ var PostcodesIO = new PostcodesIOClient();
 // eslint-disable-next-line no-var
 var renderer = require('./renderer');
 
-function outsideEngland(outcodeDetails) {
-  return !outcodeDetails.country.includes('England');
+function getCountryAsArray(country) {
+  return Array.isArray(country) ? country : [country];
 }
 
-function lookupPostcode(req, res, next) {
-  const postcode = res.locals.postcode;
+function isOutcode(postcodeDetails) {
+  return !postcodeDetails.incode;
+}
+
+function postcodeDetailsMapper(postcodeDetails) {
+  return {
+    isOutcode: isOutcode(postcodeDetails),
+    location: {
+      lat: postcodeDetails.latitude,
+      lon: postcodeDetails.longitude
+    },
+    country: getCountryAsArray(postcodeDetails.country)
+  };
+}
 
-  log.debug({ postcode }, 'lookupPostcode');
+async function lookupPostcode(req, res, next) {
+  const postcodeSearch = res.locals.postcodeSearch;
 
-  if (postcode) {
-    if (postcodeValidator.isOutcode(postcode)) {
-      log.debug('is outcode');
-      PostcodesIO.outcode(postcode, (err, outcodeDetails) => {
-        log.debug({ postcodeIOResponse: { outcodeDetails } }, 'PostcodeIO outcode response');
-        if (outcodeDetails && outsideEngland(outcodeDetails)) {
-          renderer.postcodeNotEnglish(postcode, req, res);
-        } else if (outcodeDetails) {
-          res.locals.location = { lat: outcodeDetails.latitude, lon: outcodeDetails.longitude };
-          res.locals.isOutcode = true;
-          next();
-        } else if (!err) {
-          renderer.invalidPostcodePage(postcode, req, res);
-        } else {
-          renderer.postcodeError(err, outcodeDetails, res, next);
-        }
-      });
-    } else if (postcodeValidator.isPostcode(postcode)) {
-      log.debug('is postcode');
-      PostcodesIO.lookup(postcode, (err, postcodeDetails) => {
-        log.debug({ postcodeIOResponse: { postcodeDetails } }, 'PostcodeIO postcode response');
-        if (postcodeDetails && postcodeDetails.country === 'England') {
-          res.locals.location = { lat: postcodeDetails.latitude, lon: postcodeDetails.longitude };
-          next();
-        } else if (postcodeDetails && postcodeDetails.country !== 'England') {
-          renderer.postcodeNotEnglish(postcode, req, res);
-        } else {
-          renderer.postcodeError(err, postcodeDetails, res, next);
-        }
-      });
-    } else {
-      log.debug('not valid outcode or postcode');
-      next();
+  log.debug({ postcodeSearch }, 'postcode search text');
+  if (postcodeSearch) {
+    try {
+      const postcodeDetails = await PostcodesIO.lookup(postcodeSearch);
```

Comment:
I assume istanbul and eslint were upgraded to stop it complaining about await/async keywords, as @markysoft mentioned a while ago?